xtc/src/cycles_wallet.rs

```diff
@@ -39,9 +41,10 @@ async fn call(args: CallCanisterArgs) -> Result<CallResult, String> {
         return Err("Attempted to call forward on self. This is not allowed.".to_string());
     }
 
+    let fee = compute_fee(args.cycles);
     let ledger = storage::get_mut::<Ledger>();
     ledger
-        .withdraw(&user, args.cycles)
+        .withdraw(&user, args.cycles + fee)
```

Comment:
:skull_and_crossbones:
  
this line looks innocent, but it is phrone to integer overflow and could result of Dank being hacked. If an attacker sends args.cycles = uint64.max - 2000, then args.cycles + fee will overflow and will be 0, so the attackers balance doesn't decrease and later he will receive uint64.max - 2000. 

Of course it is unlikely that our canister have that many cycles on it (impossible right now thanks to the canister have a limit on the number of cycles). However we should never have integer overflows in our smart contracts. Rust have safe math functions they throw when integer overflow or underflow happens, I think we should use those.